src/ng/compile.js

```diff
@@ -983,7 +983,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
   var bindingCache = createMap();
 
   function parseIsolateBindings(scope, directiveName, isController) {
-    var LOCAL_REGEXP = /^\s*([@&<]|=(\*?))(\??)\s*(\w*)\s*$/;
+    var LOCAL_REGEXP = /^\s*([@&<]|=(\*?))(\??)\s*([$]?\w*[$]?)\s*$/;
```

Comment:
Gotta admit I was thinking the same. But I prefered to open the PR to fix only as described in the bug. 

I'll fix it.